File: IAIso-v5.0/core/iaiso-java/iaiso-conformance/src/main/java/io/iaiso/conformance/ConformanceRunner.java
package io.iaiso.conformance;

import java.io.IOException;
import java.nio.file.Path;

/** Top-level conformance runner. */
public final class ConformanceRunner {
    private ConformanceRunner() {}

    /** Run every section against the spec at {@code specRoot}. */
    public static SectionResults runAll(Path specRoot) throws IOException {
        SectionResults s = new SectionResults();
        s.pressure.addAll(PressureRunner.run(specRoot));
        s.consent.addAll(ConsentRunner.run(specRoot));
        s.events.addAll(EventsRunner.run(specRoot));
        s.policy.addAll(PolicyRunner.run(specRoot));
        return s;
    }
}

File: IAIso-v5.0/core/iaiso-swift/Sources/IAIsoConformance/ConformanceRunner.swift
import Foundation

/// Top-level conformance runner.
public enum ConformanceRunner {

    /// Run every section against the spec at `specRoot`.
    public static func runAll(specRoot: String) throws -> SectionResults {
        var r = SectionResults()
        r.pressure = try PressureRunner.run(specRoot: specRoot)
        r.consent  = try ConsentRunner.run(specRoot: specRoot)
        r.events   = try EventsRunner.run(specRoot: specRoot)
        r.policy   = try PolicyRunner.run(specRoot: specRoot)
        return r
    }
}
